### PR TITLE
Ruff

### PR DESCRIPTION
**Pull request by hk21702**
Migrate to use ruff and replace isort, flake8 and black

### CODE REVIEW

**Comment by hk21702** [comment]
Make sure to disable flake8, black and isort for the workspace in your vscode setup. There doesn't seem to be a way to disable them in the vscode settings config file.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -18,20 +18,17 @@ jobs:
...
       - name: Super-linter
-        uses: super-linter/super-linter@v6.5.1 # x-release-please-version
+        uses: super-linter/super-linter/slim@v6.5.1 # x-release-please-version
         env:
           # To report GitHub Actions status checks
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          VALIDATE_PYTHON_BLACK: true
-          VALIDATE_PYTHON_FLAKE8: true
-          VALIDATE_PYTHON_ISORT: true
           VALIDATE_PYTHON_MYPY: true
```
[the remark this answers is not in the document]
Comment (hk21702) [reply .github/workflows/lint.yml:33]: yes. ruff isn't a replacement for mypy

File: requirements_develop.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 autoflake>=2.3.0
```
Comment (LionelColaso) [inline requirements_develop.txt:1]: image
it also replaces autoflake

**Review by LionelColaso — APPROVED** [review]
If its working then all good

**Review by twstagg — APPROVED** [review]

**Pull request merged**